File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;
import entity.Result;

import java.util.List;
import java.util.Map;

/**
 * 品牌管理的业务逻辑层接口
 */
public interface BrandService {
    //全查
    List<TbBrand> findAll();

    //分页查询所有信息，以及模糊查询
    PageResult findPageAndLikeQuery(TbBrand tbBrand, int page, int rows);

    //添加信息以及修改信息
    void saveBrandInfo(TbBrand tbBrand);

    //根据id查询
    TbBrand findOneById(Long id);

    //根据id批量删除品牌信息
    Result deleteBatchById(Long[] selectIds);

    //全查品牌信息，模板管理select2插件回显时用的
    List<Map> selectBrandOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//品牌服务层
app.service('brandService',function($http){
    //读取列表数据绑定到表单中
    this.findAll = function(){
        return  $http.get('../brand/findAll.do');
    },
        //返回分页
        this.findPage = function(searchEntity,page,rows){
            return $http.post('../brand/findPageAndLikeQuery.do?page=' + page + '&rows=' + rows,searchEntity);
        },
        //返回根据id查询的信息
        this.findOne = function(id){
            return $http.get("../brand/findOneById.do?id="+id);
        },
        //新增
        this.addBrand = function(tbbrand){
            return $http.post("../brand/saveBrandInfo.do",tbbrand);
        },
        //批量删除
        this.deleteBatchById = function (selectIds) {
            return $http.post("../brand/deleteBatchById.do",selectIds);
        },
        //全查品牌信息，模板管理select2插件回显时用的
        this.selectBrandOptionList = function(){
            return $http.get("../brand/selectBrandOptionList.do");
        }
});